### PR TITLE
Add comprehensive tests for ColumnArrayBuilder handling nested structs and mixed types

### PR DESCRIPTION
**Pull request by peterxcli**
## Summary

https://github.com/Mooncake-Labs/moonlink/pull/1296 actually has support both array of composite, nested composite, nested array in composite and multi-dim array building.

So this is a follow to test if nested array in composite works as well.

## Related Issues

Closes https://github.com/Mooncake-Labs/moonlink/issues/1228

## Changes

- Only unit tests for "nested array in composite" scenario and more edge cases testing added

## Checklist

- [x] Code builds correctly
- [x] Tests have been added or updated
- [x] Documentation updated if necessary
- [x] I have reviewed my own changes

### CODE REVIEW

**Comment by peterxcli** [comment]
cc @dentiny please take a look if you have time. Thanks you very much!

File: src/moonlink/src/row/column_array_builder.rs [hunk 1]
```diff
@@ -1234,4 +1234,238 @@ mod tests {
         assert!(field3.value(0));
         assert!(!field3.value(1));
     }
+
+    #[test]
+    fn test_column_array_builder_struct_with_list_field() {
```
Comment (dentiny) [inline src/moonlink/src/row/column_array_builder.rs]: FYI: I'm fine with the current test cases name, but you don't need to stress "column_array_builder" since rust tests are namespaced scoped.

File: src/moonlink/src/row/column_array_builder.rs [hunk 2]
```diff
@@ -1234,4 +1234,238 @@ mod tests {
...
+        use arrow::array::{ListArray, StructArray};
+
+        let struct_fields = vec![
```
Comment (dentiny) [inline src/moonlink/src/row/column_array_builder.rs:1176]: nit: for structs I usually suggest to comment out struct in rust style for easier read.

File: src/moonlink/src/row/column_array_builder.rs [hunk 3]
```diff
@@ -1234,4 +1234,238 @@ mod tests {
...
+            .unwrap();
+
+        // Verify list with values, empty list, and null
+        assert_eq!(scores_column.value(0).len(), 2);
+        assert_eq!(scores_column.value(1).len(), 0);
+        assert!(scores_column.is_null(2));
+    }
+
+    #[test]
+    fn test_column_array_builder_deeply_nested_struct() {
```
Comment (dentiny) [inline src/moonlink/src/row/column_array_builder.rs]: TBH I'm not too sure how it's different from `test_column_array_builder_nested_struct`.. but I'm ok with it.

File: src/moonlink/src/row/column_array_builder.rs [hunk 4]
```diff
@@ -1234,4 +1234,238 @@ mod tests {
...
+
+        // Verify mixed types: valid nested struct, null nested struct, and lists
+        assert!(!nested_column.is_null(0));
+        assert!(nested_column.is_null(1));
+        assert_eq!(tags_column.value(0).len(), 1);
+        assert_eq!(tags_column.value(1).len(), 1);
+    }
+
+    #[test]
+    fn test_column_array_builder_struct_with_partial_nulls() {
```
Comment (dentiny) [inline src/moonlink/src/row/column_array_builder.rs]: What's the difference with https://github.com/Mooncake-Labs/moonlink/blob/00c4819c4b9157a3f777a75e4bafdc7bbf5b8318/src/moonlink/src/row/column_array_builder.rs#L1196-L1202 ?

File: src/moonlink/src/row/column_array_builder.rs (hunk 3 above)
Comment (dentiny) [reply src/moonlink/src/row/column_array_builder.rs]: Do you mind leaving only one of them? Maybe this deeply nested one.

File: src/moonlink/src/row/column_array_builder.rs (hunk 4 above)
Comment (peterxcli) [reply src/moonlink/src/row/column_array_builder.rs]: yeah, I just merged it into the existing one. Thanks for catching this!

**Review by dentiny — APPROVED** [review]
Thanks!

**Comment by peterxcli** [comment]
Thanks for reviewing!
